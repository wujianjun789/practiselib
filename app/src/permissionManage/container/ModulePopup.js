import React,{Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import Panel from '../../components/Panel';
import PanelFooter from '../../components/PanelFooter';
import '../../../public/styles/permissionManage.less';
import {IsExitInArray,getObjectByKeyObj,spliceInArray} from '../../util/algorithm';
import {FormattedMessage} from 'react-intl';

export default class ModulePopup extends Component{
    constructor(props){
        super(props);
        const {data=[]} = this.props;
        this.state = {
            moduleList:data
        }
        this.onCancel = this.onCancel.bind(this);
        this.onConfirm = this.onConfirm.bind(this);
        this.moduleAdd = this.moduleAdd.bind(this);
        this.moduleDelete =this.moduleDelete.bind(this);
    }

    onCancel(){
        this.props.overlayerHide();
    }

    onConfirm(){
        let data = {
            modules:this.state.moduleList,
            id:this.props.id
        }
        this.props.onConfirm(data,true);
        this.props.overlayerHide();
    }

    moduleDelete(item){
        let {moduleList} = this.state;
        spliceInArray(moduleList,item);
        this.setState({moduleList:moduleList});
    }

    moduleAdd(item){
        let {moduleList} = this.state;
        moduleList.push(item);
        this.setState({moduleList:moduleList});        
    }

    render() {
        let {className = '',title = '',modules,id,data=[]} = this.props;
        let {moduleList} = this.state;
        let footer = <PanelFooter funcNames={['onCancel','onConfirm']} btnTitles={['button.cancel','button.confirm']} btnClassName={['btn-default', 'btn-primary']} btnDisabled={[false, false]} onCancel={this.onCancel} onConfirm={this.onConfirm}/>;
        return (
            <Panel className={className} title = {title} footer = {footer} closeBtn = {true} closeClick = {this.onCancel}>
                <div className = 'form-group row module-per'>
                    <label className="control-label"><FormattedMessage id='permission.module'/>:</label>
                    <div className="module-content">
                        <div className='all-module-list'>
                            <ul>
                            {
                                modules.map((item,index)=>{
                                    return <li key = {index}>
                                        {    
                                            IsExitInArray(moduleList,item.key)?<span className='list-node-add'><FormattedMessage id='permission.added'/></span>:<span className='glyphicon glyphicon-plus' onClick={()=>this.moduleAdd(item.key)}></span>
                                        }
                                        {<FormattedMessage id={item.title}/>}                                    
                                    </li>
                                })
                            }
                            </ul>
                        </div>
                        <div className='module-list'>
                            <ul>
                                {
                                    moduleList.map((item,index)=>{
                                        return <li key = {index}>
                                            <span className="icon-table-delete" onClick={()=>this.moduleDelete(item)}></span>
                                            {<FormattedMessage id={getObjectByKeyObj(modules,'key',item)['title']}/>}
                                        </li>
                                    })
                                }
                            </ul>
                        </div>
                    </div>
                </div>
            </Panel>
        )
    }
}