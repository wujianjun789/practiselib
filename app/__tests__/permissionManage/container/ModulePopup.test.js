import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import ModulePopup from '../../../src/permissionManage/container/ModulePopup';
// import {Provider} from 'react-redux';
import Immutable from 'immutable';
// import configureStore from '../../../src/store/configureStore';

describe('<ModulePopup',()=>{
    // const store = configureStore();
    const className = 'user-module-edit-popup';
    const title = '模块权限管理';
    const modules = [
        {key: 'asset', title: '资产管理', link: '/assetManage/manage'},
        {key: 'permission', title: '权限管理', link: '/permissionManage'},
        {key: 'maintenance', title: '系统运维', link: '/systemOperation/gateway'},
        {key: 'control', title: '系统控制', link: '/'},
        {key: 'report', title: '报表管理', link: '/'},
        {key: 'publish', title: '媒体发布', link: '/'},
        {key: 'visual', title: '可视化', link: '/'},
        {key: 'domain', title: '域管理', link: '/domainManage/domainEdit'}
    ];
    const rowModules = ['permission','maintenance'];
    it('render normal',()=>{
        const cmp = shallow(<ModulePopup className={className} title={title} modules={modules} data = {rowModules}/>);
        const container = cmp.find(`.${className}`);
        expect(container.length).toBe(1);

        const allModule = container.find('.all-module-list');
        expect(allModule.length).toBe(1);
        const allModuleLi = allModule.find('li');
        expect(allModuleLi.length).toBe(modules.length);

        const moduleList = container.find('.module-list');
        expect(moduleList.length).toBe(1);
        const moduleLi = moduleList.find('li');
        expect(moduleLi.length).toBe(rowModules.length);
    })

    it('snapshot', () => {
        const cmp = renderer.create(<ModulePopup className={className} title={title} modules={modules} data = {rowModules}/>);
        const tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    });
})
