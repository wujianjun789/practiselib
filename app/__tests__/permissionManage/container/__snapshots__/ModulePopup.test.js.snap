exports[`<ModulePopup snapshot 1`] = `
<div
  className="modal-content user-module-edit-popup">
  <div
    className="modal-header">
    <button
      aria-label="Close"
      className="close"
      data-dismiss="modal"
      onClick={[Function]}
      type="button">
      <span
        aria-hidden="true">
        ×
      </span>
    </button>
    <h4
      className="modal-title">
      模块权限管理
    </h4>
  </div>
  <div
    className="modal-body">
    <div
      className="form-group row module-per">
      <label
        className="control-label">
        模块权限:
      </label>
      <div
        className="module-content">
        <div
          className="all-module-list">
          <ul>
            <li>
              <span
                className="glyphicon glyphicon-plus"
                onClick={[Function]} />
              资产管理
            </li>
            <li>
              <span
                className="list-node-add">
                已添加
              </span>
              权限管理
            </li>
            <li>
              <span
                className="list-node-add">
                已添加
              </span>
              系统运维
            </li>
            <li>
              <span
                className="glyphicon glyphicon-plus"
                onClick={[Function]} />
              系统控制
            </li>
            <li>
              <span
                className="glyphicon glyphicon-plus"
                onClick={[Function]} />
              报表管理
            </li>
            <li>
              <span
                className="glyphicon glyphicon-plus"
                onClick={[Function]} />
              媒体发布
            </li>
            <li>
              <span
                className="glyphicon glyphicon-plus"
                onClick={[Function]} />
              可视化
            </li>
            <li>
              <span
                className="glyphicon glyphicon-plus"
                onClick={[Function]} />
              域管理
            </li>
          </ul>
        </div>
        <div
          className="module-list">
          <ul>
            <li>
              <span
                className="icon-table-delete"
                onClick={[Function]} />
              权限管理
            </li>
            <li>
              <span
                className="icon-table-delete"
                onClick={[Function]} />
              系统运维
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modal-footer">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      取消
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button">
      确认
    </button>
  </div>
</div>
`;
